components/carousel/script.js

```diff
@@ -0,0 +1,368 @@
+import { Components } from 'coherent-gameface-components';
+import template from './template.html';
+
+const components = new Components();
+const BaseComponent = components.BaseComponent;
+
+const CONTENT_WRAPPER_SELECTOR = '.guic-carousel-content-wrapper';
+const DOTS_CONTAINER_SELECTOR = '.guic-carousel-dots';
+const DOT_SELECTOR = '.guic-carousel-dot';
+const CAROUSEL_SELECTOR = '.guic-carousel';
+const NEXT_BTN_SELECTOR = '.guic-carousel-btn-next';
+const PREV_BTN_SELECTOR = '.guic-carousel-btn-prev';
+const NEXT_ARROW_SELECTOR = '.guic-carousel-right-arrow';
+const PREV_ARROW_SELECTOR = '.guic-carousel-left-arrow';
+
+/**
+ * Class description
+ */
+class Carousel extends BaseComponent {
+    /* eslint-disable-next-line require-jsdoc */
+    constructor() {
+        super();
+        this.template = template;
+        this.init = this.init.bind(this);
+
+        this._pageSize = 2;
+        this._navArrowStepSize = 1;
+        this.current = 0;

```

Comment:
Can we rename this (+usages) to `this.currentIndex = 0;` or `this.currentItemIndex = 0;`.